subxt/Cargo.toml

```diff
@@ -22,19 +22,20 @@ integration-tests = []
 
 # Jsonrpsee if the default RPC provider used in Subxt. However, it can be
 # swapped out for an alternative implementation, and so is optional.
-jsonrpsee = ["dep:jsonrpsee"]
+jsonrpsee-ws = ["jsonrpsee/async-client", "jsonrpsee/client-ws-transport"]
+jsonrpsee-web = ["jsonrpsee/async-wasm-client", "jsonrpsee/client-web-transport"]
 
 [dependencies]
 bitvec = { version = "1.0.0", default-features = false, features = ["alloc"] }
 codec = { package = "parity-scale-codec", version = "3.0.0", default-features = false, features = ["derive", "full", "bit-vec"] }
 scale-info = { version = "2.0.0", features = ["bit-vec"] }
 scale-value = "0.6.0"
 scale-decode = "0.4.0"
-futures = "0.3.13"
+futures = { version = "0.3.13", default-features = false }
 hex = "0.4.3"
-jsonrpsee = { version = "0.15.1", features = ["async-client", "client-ws-transport", "jsonrpsee-types"], optional = true }
+jsonrpsee = { version = "0.16", optional = true, features = ["jsonrpsee-types"] }
```

Comment:
Is it possible to still keep `jsonrpsee` as an optional dependency? For anyone implementing a custom client who doesn't need it.

Reply:
yes, good catch

Reply:
Oh yes sorry the optional moved before the features :grin: